packages/api/cms-api/src/redirects/redirects.resolver.ts

```diff
@@ -121,15 +121,18 @@ export function createRedirectsResolver({
             @Args("scope", { type: () => Scope, defaultValue: hasNonEmptyScope ? undefined : {} }) scope: typeof Scope,
             @Args("source", { type: () => String }) source: string,
         ): Promise<boolean> {
-            const redirect = await this.repository.findOne({ source, ...(hasNonEmptyScope ? { scope: nonEmptyScopeOrNothing(scope) } : {}) });
-            return redirect === null;
+            return this.redirectService.isRedirectSourceAvailable(source, nonEmptyScopeOrNothing(scope));
         }
 
         @Mutation(() => Redirect)
         async createRedirect(
             @Args("scope", { type: () => Scope, defaultValue: hasNonEmptyScope ? undefined : {} }) scope: typeof Scope,
             @Args("input", { type: () => RedirectInput }) input: RedirectInputInterface,
         ): Promise<RedirectInterface> {
+            if (!(await this.redirectService.isRedirectSourceAvailable(input.source, scope))) {
+                throw new CometValidationException("Validation failed");
+            }
```

Comment:
@johnnyomair Is this the best way to do this validation?

I couldn't do it in a decorator on the `RedirectInput` since I need the `scope` to determine if the `source` is occupied or not

Reply:
Yes, I think this is the best way. 

@nsams what do you think?

Reply:
:+1: